codalab/model/bundle_model.py

```diff
@@ -763,6 +806,45 @@ def save_bundle(self, bundle):
                 self.do_multirow_insert(connection, cl_bundle_metadata, metadata_values)
                 bundle.id = result.lastrowid
 
+    def save_bundle_rest(self, bundle_info):
+        """
+        Save a bundle. On success, sets the bundle's id from the result.
+        Bundle info should be validated prior to calling this method.
+        """
+        dependency_values = bundle_info.pop('dependencies')
+        metadata_values = bundle_info.pop('metadata')
+
+        # Check to see if bundle is already present, as in a local 'cl cp'
+        if not self.batch_get_bundles(uuid=bundle_info['uuid']):
+            with self.engine.begin() as connection:
+                connection.execute(cl_bundle.insert().values(bundle_info))
+                self.do_multirow_insert(connection, cl_bundle_dependency, dependency_values)
+                self.do_multirow_insert(connection, cl_bundle_metadata, metadata_values)
+
+    def update_bundle_rest(self, bundle_update):
```

Comment:
Deprecating all the ORM objects was my original plan, since we had discussed how having the ORM objects and Schema and dicts was redundant. So I wanted to do all the validation through the Schemas. However I didn't realize how much code depended on the ORM objects, and also I now agree with you on the terribleness of copying code.

@klopyrev I am now thinking of reverting back to using the ORM objects and the original BundleModel methods, and keeping validation code inside the ORM objects. What do you think?